File: radio/app/src/main/java/com/example/radio/MainActivity.java
package com.example.radio;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText txt1, txt2;
    TextView tvre;
    Button tbope;
    RadioButton r1,r2;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        txt1 = (EditText) findViewById(R.id.txt1);
        txt2 = (EditText) findViewById(R.id.txt2);
        tvre = (EditText) findViewById(R.id.tvres);
        r1=(RadioButton) findViewById(R.id.r1);
        r2=(RadioButton) findViewById(R.id.r2);

        tbope = (Button) findViewById(R.id.btnres);
        //metodo onclick
        tbope.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /*String val1 = txt1.getText().toString();
                String val2 = txt2.getText().toString();
                int op1 = Integer.parseInt(val1);
                int op2 = Integer.parseInt(val2);*/
                if(r1.isChecked() == true){
                    /*int res = op1 +op2;
                    String strinres= String.valueOf(res);
                    tvre.setText(strinres);*/
                    tvre.setText(String.valueOf(Integer.parseInt(txt1.getText().toString())+Integer.parseInt(txt2.getText().toString())));
                }
            }
        });
    }
}

File: miPasosKotlin/app/src/main/java/com/example/mipasoskotlin/MainActivity.kt
package com.example.mipasoskotlin

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.mipasoskotlin.models.Productos

class MainActivity : AppCompatActivity() {
    lateinit var btnCalcular: Button
    lateinit var txtPrecio: EditText
    lateinit var tvResul: TextView
    lateinit var spLista:Spinner

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        //codigo
        cargarR()
        estadoOnclick()

        //cargar listad e datos en spinner
        val listaPaises = arrayOf("USA", "BOL", "ESP")
        val adaptador1 = ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, listaPaises)
        spLista.adapter = adaptador1


    }
    fun cargarR (){
        btnCalcular = findViewById(R.id.btnCalcularIVA)
        txtPrecio = findViewById(R.id.txtProducto)
        tvResul = findViewById(R.id.tvResultado)
        spLista = findViewById(R.id.spPaises)
    }

    fun estadoOnclick(){
        btnCalcular.setOnClickListener(){

            val laptop = Productos("laptop", txtPrecio.text.toString().toDouble())
            //val datosRe : Double = laptop.calIVA()

            when(spLista.selectedItem.toString()){
                "USA"-> tvResul.text = laptop.calIVA(0.03).toString()
                "BOL"-> tvResul.text = laptop.calIVA(0.13).toString()
                "ESP"-> tvResul.text = laptop.calIVA(0.05).toString()
            }

        }
    }



}